src/Compilers/CSharp/Portable/Binder/RefSafetyAnalysis.cs

```diff
@@ -405,8 +491,15 @@ private void AddLocalScopes(LocalSymbol local, uint refEscapeScope, uint valEsca
                     CallingMethodScope;
             }
 
+            Debug.Assert(_localEscapeScopes?.ContainsKey(local) != true);
+
+            AddOrSetLocalScopes(local, refEscapeScope, valEscapeScope);
+        }
+
+        private void AddOrSetLocalScopes(LocalSymbol local, uint refEscapeScope, uint valEscapeScope)
+        {
             _localEscapeScopes ??= new Dictionary<LocalSymbol, (uint RefEscapeScope, uint ValEscapeScope)>();
-            _localEscapeScopes.Add(local, (refEscapeScope, valEscapeScope));
+            _localEscapeScopes[local] = (refEscapeScope, valEscapeScope);
```

Comment:
This used to only add only, but now can also set. What's the scenario where we need to set the scopes of a local more than once?

Reply:
> What's the scenario where we need to set the scopes of a local more than once?

I don't think there are valid scenarios where the local is added twice or set twice. A local will be added and set separately though - the local is added when visiting the containing block, and set when visiting the initializer. The callers assert that the local is missing or not missing depending on the context. But the change here, from add to set, was from [https://github.com/dotnet/roslyn/pull/67092](https://github.com/dotnet/roslyn/pull/67092) to allow this to be more forgiving in the unexpected case where the local is added twice.